x/farming/keeper/reward.go

```diff
@@ -159,3 +147,117 @@ func (k Keeper) Harvest(ctx sdk.Context, farmerAcc sdk.AccAddress, stakingCoinDe
 
 	return nil
 }
+
+type DistributionInfo struct {
+	Plan   types.PlanI
+	Amount sdk.Coins
+}
+
+func (k Keeper) DistributionInfos(ctx sdk.Context) []DistributionInfo {
+	farmingPoolBalances := make(map[string]sdk.Coins)   // farmingPoolAddress => sdk.Coins
+	distrCoins := make(map[string]map[uint64]sdk.Coins) // farmingPoolAddress => (planId => sdk.Coins)
+
+	plans := make(map[uint64]types.PlanI)
+	for _, plan := range k.GetAllPlans(ctx) {
+		// Filter plans by their start time and end time.
+		if !plan.GetStartTime().After(ctx.BlockTime()) && plan.GetEndTime().After(ctx.BlockTime()) {
+			plans[plan.GetId()] = plan
+		}
+	}
+
+	for _, plan := range plans {
+		farmingPoolAcc := plan.GetFarmingPoolAddress()
+		farmingPool := farmingPoolAcc.String()
+
+		balances, ok := farmingPoolBalances[farmingPool]
+		if !ok {
+			balances = k.bankKeeper.GetAllBalances(ctx, farmingPoolAcc)
+			farmingPoolBalances[farmingPool] = balances
+		}
+
+		dc, ok := distrCoins[farmingPool]
+		if !ok {
+			dc = make(map[uint64]sdk.Coins)
+			distrCoins[farmingPool] = dc
+		}
+
+		switch plan := plan.(type) {
+		case *types.FixedAmountPlan:
+			dc[plan.GetId()] = plan.EpochAmount
+		case *types.RatioPlan:
+			dc[plan.GetId()], _ = sdk.NewDecCoinsFromCoins(balances...).MulDecTruncate(plan.EpochRatio).TruncateDecimal()
+		}
+	}
+
+	var distrInfos []DistributionInfo
+	for farmingPool, coins := range distrCoins {
+		totalCoins := sdk.NewCoins()
+		for _, amt := range coins {
+			totalCoins = totalCoins.Add(amt...)
+		}
+
+		balances := farmingPoolBalances[farmingPool]
+		fmt.Println("farmingPool=", farmingPool, "balances=", balances, "totalCoins=", totalCoins)
```

Comment:
This line seems like you used it for the test.

Reply:
Yes right. I'll remove it